### PR TITLE
Add the load user actions.

### PR DESCRIPTION
**Pull request by rhaiamz**
[AB#467](https://dev.azure.com/plantanapp/0e4d84fe-a4cf-4594-b253-faab0de412c9/_workitems/edit/467)

### CODE REVIEW

File: docs/Actions/load-users-sql.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+| Parameter     | Description                           | Supports Variables |
+|---------------|---------------------------------------|--------------------|
+| SQL Query | The SQL query will define what users will be loaded into context from the database. The SQL result must be an user ID, and user email, or a user name. | Yes |
+
+
+## `Examples`
+
+### `1. Load a list of users and grant the a role`
+
+The actions below will load the user with the identifier (user ID) '0' into the context and then grant him the role `Subscribers` with an indefinite expiration date.
```
Comment (bogdan-litescu) [inline docs/Actions/load-users-sql.md:36]: This description seems copy-past as is not accurate.

File: docs/Actions/load-users-sql.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+---
+id: LoadUsersSql
+title: Load Users From SQL
+sidebar_label: Load Users From SQL
+---
+
+An action that allows altering the contextual users which all the subsequent actions use. This action will not alter the context user (logged in user).
```
Comment (bogdan-litescu) [inline docs/Actions/load-users-sql.md:7]: Feel that the clarification is not sufficient between what context.User and context.Users mean. They're both in the context.

File: docs/Actions/load-users-sql.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+When used in conjunction with actions such as [Grant Role](/docs/Actions/grant-role.md), the `Grant Role` action will grant the role specified to all users loaded by this action.
+
+## `Typical Use Cases`
+
+- Loading the users before granting the necessary roles
+- Loading users before sending a mass email
+
+## `Don't use it to`
+
+- Alter the currently logged in context user, use [Load User](/docs/actions/load-user.md) instead
```
Comment (bogdan-litescu) [inline docs/Actions/load-users-sql.md:18]: Does the context user change the currently logged in user?

File: docs/Actions/load-user.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+---
+id: LoadUser
+title: Load User
+sidebar_label: Load User
+---
+
+An action that allows altering the contextual users which all the subsequent actions use. This action supports loading multiple users by providing multiple user identifiers. The last user identifier provided will also be used as the context user under which all actions execute (logged in user).
```
[the remark this answers is not in the document]
Comment (bogdan-litescu) [reply docs/Actions/load-user.md:7]: As a first time user, do I know what "contextual users" mean. 
Let's explain what we know
- by default, the actions refer to to the user that's currently logged in
- and that that user can be changed just for the duration of the workflow using these actions

File: docs/Actions/load-user.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+- [Authorize User](/docs/Actions/authorize-user.md)
+- [Unauthorize User](/docs/Actions/unauthorize-user.md)
+- [Delete User](/docs/Actions/delete-user.md)
+- [Send Email](/doc/Actions/send-email.md).
+
+When used in conjunction with actions such as [Grant Role](/docs/Actions/grant-role.md), the `Grant Role` action will grant the role specified to all users loaded by this action.
+
+## `Typical Use Cases`
+
+- Impersonating an user
```
Comment (radun2) [inline docs/Actions/load-user.md:22]:
```suggestion
- Update another User profile
```

File: docs/Actions/load-user.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+- Loading the user before granting the necessary roles
+- Loading users before sending a mass email
+
+## `Don't use it to`
+
+- Load multiple users by a specific condition, use [Load Users From SQL](/docs/actions/load-users-sql.md) instead
+
+## `Related Actions`
+
```
Comment (radun2) [inline docs/Actions/load-user.md:31]:
```suggestion
- [Update User Profile](/docs/Actions/update-user-profile.md) - updates user profile properties 
```

File: docs/Actions/load-user.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    "ActionType": "LoadUser",
+    "Description": "Load the user into the context",
+    "Condition": null,
+    "Parameters": {
+        "Id": "john.doe@gmail.com",
+        "Portal": ""
+    }
+}
+```
+
```
Comment (radun2) [inline docs/Actions/load-user.md:60]: Maybe we can add an Update User Profile Example for FirstName and LastName, Something simple

File: docs/Actions/load-user.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
+---
+id: LoadUser
+title: Load User
+sidebar_label: Load User
+---
+
+An action that allows altering the user which the subsequent actions refer to. The user change affects all actions after the `Load User` action and after all actions have finished executing the user will be reverted back to the initial one. By default, all actions that need an user for their execution will refer to the currently logged in user.
```
Comment (radun2) [inline docs/Actions/load-user.md:7]:
```suggestion
An action that allows altering the user which the subsequent actions refer to. The user change affects all actions after the `Load User` action and after all actions have finished executing the user will be reverted back to the initial one (eg: the logged-in user). By default, all actions that need a user for their execution will refer to the currently logged in user.
```

File: docs/Actions/load-user.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+## `Input Parameter Reference`
+
+| Parameter     | Description                           | Supports Variables |
+|---------------|---------------------------------------|--------------------|
+| User Identifier | The identifier which will indicate which user to load into context. You can use the user ID, username or the email as identifiers. <br/> You can load multiple users by providing multiple values separated by a comma. | Yes |
+| Portal ID | The identifier which will specify in which portal to look for the user indicated by the 'User Identifier'. If left blank, the action will search for the user across all portals. Valid values are either the portal ID or the portal name. | Yes |
+
+## `Examples`
+
+### `1. Load an user and grant it the Subscribers role`
```
Comment (radun2) [inline docs/Actions/load-user.md:44]:
```suggestion
### `1. Load a user and grant it the Subscribers role`
```

File: docs/Actions/load-users-sql.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+## `Related Actions`
+
+- [Grant Role](/docs/Actions/grant-role.md) - grants the users in the context the specified roles
+- [Send Email](/doc/Actions/send-email.md) - sends an email to the users loaded in the context
+
+## `Input Parameter Reference`
+
+| Parameter     | Description                           | Supports Variables |
+|---------------|---------------------------------------|--------------------|
+| SQL Query | The SQL query will define what users will be loaded into context from the database. The SQL result must be an user ID, and user email, or a user name. | Yes |
```
Comment (radun2) [inline docs/Actions/load-users-sql.md:36]:
```suggestion
| SQL Query | The SQL query will define what users will be loaded into context from the database. The SQL result must be a user ID, and user email, or a user name. | Yes |
```

File: docs/Actions/load-user.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+This action also supports loading multiple users by providing multiple user identifiers. The last user identifier provided will also be used as the context user under which all actions execute. Loading multiple users creates a hidden context entity that is leveraged by the following actions which support this:
+
+- [Grant Role](/docs/Actions/grant-role.md)
+- [Revoke Role](/docs/Actions/revoke-role.md)
+- [Authorize User](/docs/Actions/authorize-user.md)
+- [Unauthorize User](/docs/Actions/unauthorize-user.md)
+- [Delete User](/docs/Actions/delete-user.md)
+- [Send Email](/doc/Actions/send-email.md).
+
+When used in conjunction with actions such as [Grant Role](/docs/Actions/grant-role.md), the `Grant Role` action will grant the role specified to all users loaded by this action.
```
Comment (bogdan-litescu) [inline docs/Actions/load-user.md:18]:
```suggestion
When used in conjunction with the actions above, the logic applies to all users that have been loaded into the context by the `Load User` action. For example, the `Grant Role` action will assign the specified role to each and every user loaded into the context.
```

File: docs/Actions/load-user.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+
```
Comment (bogdan-litescu) [inline docs/Actions/load-user.md:19]:
```suggestion

Calling the `Load User` action the second time inside the same workflow will replace the user or the list of users in the context. It never appends to the list that is already loaded.

```

Comment (rhaiamz) [reply docs/Actions/load-user.md:21]: This is not the actual behavior, I committed this change with the proper explanation.

File: docs/Actions/load-user.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+- [Grant Role](/docs/Actions/grant-role.md)
+- [Revoke Role](/docs/Actions/revoke-role.md)
+- [Authorize User](/docs/Actions/authorize-user.md)
+- [Unauthorize User](/docs/Actions/unauthorize-user.md)
+- [Delete User](/docs/Actions/delete-user.md)
+- [Send Email](/doc/Actions/send-email.md).
+
+When used in conjunction with the actions above, the logic applies to all users that have been loaded into the context by the `Load User` action. For example, the `Grant Role` action will assign the specified role to each and every user loaded into the context.
+
+Calling the `Load User` action the second time inside the same workflow will replace the context user that the actions use and add to the list of context users(the Users entity) that actions such as `Grant Role` leverage.
```
Comment (bogdan-litescu) [inline docs/Actions/load-user.md:20]:
```suggestion
Calling the `Load User` action the second time inside the same workflow will replace the context user that the actions use and add to the list of context users (the Users entity) that actions such as `Grant Role` leverage.
```

File: docs/Actions/load-user.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+- [Grant Role](/docs/Actions/grant-role.md)
+- [Revoke Role](/docs/Actions/revoke-role.md)
+- [Authorize User](/docs/Actions/authorize-user.md)
+- [Unauthorize User](/docs/Actions/unauthorize-user.md)
+- [Delete User](/docs/Actions/delete-user.md)
+- [Send Email](/doc/Actions/send-email.md).
+
+When used in conjunction with the actions above, the logic applies to all users that have been loaded into the context by the `Load User` action. For example, the `Grant Role` action will assign the specified role to each and every user loaded into the context.
+
+Calling the `Load User` action the second time inside the same workflow will replace the context user that the actions use and add to the list of context users (the Users entity) that actions such as `Grant Role` leverage. If a user is already present in the Users entity he will not be added again.
```
Comment (bogdan-litescu) [inline docs/Actions/load-user.md:20]: Avoid "he" or "she" as it discriminates. In this case "it" works. Or, prefer plural, as "they" is gender-neutral.

**Pull request merged**